internal/kafka/internal/services/kas_fleetshard_operator_addon.go

```diff
@@ -122,12 +126,16 @@ func (o *kasFleetshardOperatorAddon) ReconcileParameters(cluster api.Cluster) (P
 	}
 }
 
-func (o *kasFleetshardOperatorAddon) getAddonParams(cluster api.Cluster) ([]types.Parameter, *errors.ServiceError) {
-	acc, pErr := o.provisionServiceAccount(cluster.ClusterID)
-	if pErr != nil {
-		return nil, errors.GeneralError("failed to create service account for cluster %s due to error: %v", cluster.ClusterID, pErr)
+func (o *kasFleetshardOperatorAddon) getAddonParams(cluster *api.Cluster) ([]types.Parameter, *errors.ServiceError) {
+	var acc *api.ServiceAccount
+	if cluster.ClientID == "" {
```

Comment:
```suggestion
	if cluster.ClientID == "" || cluster.ClientSecret == "" {
```
let's check for the two of them.